File: CPP/Day29.cpp
/*

Day 29:	Write a program that reads in a file and prints out the number of lines, words, and characters in the file.

*/

//Program
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

int main() {
    string filename;
    cout << "Enter filename: ";  
    cin >> filename;

    ifstream file(filename);
    if (!file.is_open()) {
        cerr << "Failed to open file: " << filename << endl;
        return 1;
    }

    int line_count = 0, word_count = 0, char_count = 0;
    string line;
    while (getline(file, line)) {
        line_count++;
        char_count += line.length();
        bool in_word = false;
        for (char c : line) {
            if (isspace(c)) {
                if (in_word) {
                    word_count++;
                    in_word = false;
                }
            } else {
                in_word = true;
            }
        }
        if (in_word) {
            word_count++;
        }
    }
    file.close();

    cout << "Lines: " << line_count << endl;
    cout << "Words: " << word_count << endl;
    cout << "Characters: " << char_count << endl;

    return 0;
}


/*

Input and Output:

Enter filename: IODump.txt
Lines: 5
Words: 24
Characters: 128

*/

File: Python/Day29.py
""" 

Day 29:	Write a program that reads in a file and prints out the number of lines, words, and characters in the file.


"""

#Program
number_of_words = 0
number_of_lines = 0
number_of_characters = 0

with open("data.txt", 'r') as file:
    for l in file:
      number_of_words += len(l.split())
      number_of_lines += 1
      number_of_characters = len(l)

print("No of words: ", number_of_words)
print("No of lines: ", number_of_lines)
print("No of characters: ", number_of_characters)





"""

Input and Output: 

Enter file name: IODump.txt
Number of words in text file:  26
Number of lines in text file:  5
Number of characters in text file:  107
Number of spaces in text file:  21

"""